framework/src/Volo.Abp.Swashbuckle/Microsoft/Extensions/DependencyInjection/AbpSwaggerGenServiceCollectionExtensions.cs

```diff
@@ -64,19 +62,64 @@ public static class AbpSwaggerGenServiceCollectionExtensions
 
                     options.AddSecurityRequirement(new OpenApiSecurityRequirement
                     {
+                        {
+                            new OpenApiSecurityScheme
                             {
-                                new OpenApiSecurityScheme
+                                Reference = new OpenApiReference
                                 {
-                                    Reference = new OpenApiReference
-                                    {
-                                        Type = ReferenceType.SecurityScheme,
-                                        Id = "oauth2"
-                                    }
-                                },
-                                Array.Empty<string>()
-                            }
+                                    Type = ReferenceType.SecurityScheme,
+                                    Id = "oauth2"
+                                }
+                            },
+                            Array.Empty<string>()
+                        }
+                    });
+
+                    setupAction?.Invoke(options);
+                });
+    }
+
+    public static IServiceCollection AddAbpSwaggerGenWithOidc(
+        this IServiceCollection services,
+        [NotNull] string authority,
+        Action<SwaggerGenOptions> setupAction = null,
+        List<string> flows = null,
+        string openIdConnectDiscoveryEndpoint = "/.well-known/openid-configuration")
+    {
+        var openIdConnectDiscoveryUrl = new Uri($"{authority.TrimEnd('/')}{openIdConnectDiscoveryEndpoint.EnsureStartsWith('/')}");
```

Comment:
If we use a different endpoint for metadata, it will end up something like:
`https://localhost:44301/https://myabp.com/.well-known/openid-configuration`

If null, it should be: `new Uri($"{authority.TrimEnd('/')}{openIdConnectDiscoveryEndpoint.EnsureStartsWith('/')}"`
otherwise it should be the new URL we pass, like: `https://myabp.com/.well-known/openid-configuration`